rust/src/ssh/ssh.rs

```diff
@@ -156,20 +160,32 @@ impl SSHState {
             match parser::ssh_parse_record(input) {
                 Ok((rem, head)) => {
                     SCLogDebug!("SSH valid record {}", head);
-                    input = rem;
-                    if head.msg_code == SSH_MSG_NEWKEYS {
-                        hdr.flags = SSHConnectionState::SshStateFinished;
-                        if ohdr.flags >= SSHConnectionState::SshStateFinished {
-                            unsafe {
-                                AppLayerParserStateSetFlag(
-                                    pstate,
-                                    APP_LAYER_PARSER_NO_INSPECTION
+                    match head.msg_code {
+                        parser::MessageCode::SshMsgKexinit if unsafe {SSHHasshIsEnabled()} => {
+                        	match parser::ssh_parse_key_exchange(&input[SSH_RECORD_HEADER_LEN..]) {
+                                Ok((_, key_exchange)) => {
+                                    key_exchange.generate_hassh(&mut hdr.hassh_string, &mut hdr.hassh, &resp);
+                                }
+                                Err(_) => { }
```

Comment:
should we set an event here or something? If this is not used, I would suggest using `if let` here:
```rust
if let Ok((_, key_exchange)) = parser::ssh_parse_key_exchange(&input[SSH_RECORD_HEADER_LEN..]) {
   key_exchange.generate_hassh(&mut hdr.hassh_string, &mut hdr.hassh, &resp);
}
```

Reply:
yes, `if let` is cleaner for this use case.